### ISSUE TITLE
Can't find .wast or .abi file - WASM compiler error

### ISSUE THREAD

**Issue by danwadeloo** [EOSIO/eos#546]
Hi,

I'm stuck trying to setup local nodes and mine on the same blockchain. I get stuck at the following command to setup smart contracts:
./eosc set contract currency ../../contracts/currency/currency.wast ../../contracts/currency/currency.abi

This .wast and .abi file don't exist, and I think its because the WASM compiler wasn't properly installed and run. I've tried reinstalling WASM and restarting the blockchain, but it still doesn't work. 

Anyone know how to get past this?

Thanks!

**Comment by asvirido** [comment]
Do you have currency.abi and currency.wast?

**Comment by danwadeloo** [comment]
No I don't. Could you give them to me if you have them?

**Comment by asvirido** [comment]
@danwadeloo 
it is unlikely to help

**Comment by asvirido** [comment]
send the file log.txt 
```
cd ~/eos/build 
cmake -DBINARYEN_BIN=~/binaryen/bin -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib .. >> log.txt
make -j2  >> log.txt
```

**Comment by danwadeloo** [comment]
I get this when running the command:
CMake Error: The source directory "/home/daniel" does not appear to contain CMakeLists.txt.

**Comment by danwadeloo** [comment]
Attached to this post is the log.txt file

**Comment by danwadeloo** [comment]
[log.txt](https://github.com/EOSIO/eos/files/1364261/log.txt)

**Comment by danwadeloo** [comment]
Here is the log.txt file from running that command!

**Comment by asvirido** [comment]
@danwadeloo 
-- No WASM compiler could be found. 

This can help
echo "export BINARYEN_ROOT=~/binaryen" >> ~/.bash_profile
echo "export WASM_LLVM_CONFIG=~/wasm-compiler/llvm/bin/llvm-config" >> ~/.bash_profile
echo "export LLVM_DIR=/usr/local/Cellar/llvm/4.0.1/lib/cmake/llvm" >> ~/.bash_profile
source ~/.bash_profile

**Comment by asvirido** [comment]
Try it now

**Comment by tlightsky** [comment]
i meet this too, in dawin-v1.0.0, MacOS Sierra(all env setup ok)

**Comment by asvirido** [comment]
@tlightsky 
make log.txt file

**Comment by tlightsky** [comment]
@asvirido in mac, it won't recognize the env i setup. 
i manually write the env to the `CMakeLists.txt` file.
it works then.

strangely i met the problem under ubuntu now,
maybe could only change the `CMakeLists.txt` file. again

**Comment by tlightsky** [comment]
@asvirido it seems need to setup `WASM_CLANG,WASM_LLC,WASM_LLVM_LINK` all 4 env var:
https://github.com/EOSIO/eos/blob/master/CMakeModules/wasm.cmake#L56

**Comment by tlightsky** [comment]
maybe i should use the build full rather than setup env by myself

**Comment by danwadeloo** [comment]
@asvirido 
I tried running the commands you suggested, but still get the same error (.wast and .abi not found)

**Comment by danwadeloo** [comment]
Well, I get this error:
File does not exist: ../../contracts/currency/currency.wast
ERROR: ValidationError: wast-file=../../contracts/currency/currency.wast
Create or update the contract on an account
Usage: ./eosc set contract [OPTIONS] account wast-file [abi-file]

**Comment by danwadeloo** [comment]
@tlightsky  
Did using this work?
./build.sh ubuntu full

**Comment by tlightsky** [comment]
@danwadeloo not for mac(edit the file manually), yes for ubuntu

**Comment by danwadeloo** [comment]
@tlightsky How did you get it to work with ubuntu ?
I still can't find the .wast file when I try to run a test node.

**Comment by tlightsky** [comment]
@danwadeloo you must install those dependencies : https://github.com/EOSIO/eos#manual-installation-of-the-dependencies
or it won't compile wast contract files

**Comment by comtechnet** [comment]
There is definitely an issue here with the MacOS Sierra 10.12.6 automated build script.

Just like with some of the other folks here - I ran the script, etc. and I got the .abi files,
but i did not get the .wast files - so the example currency contract walkthrough fails with the error that eosc can not find the .wast file

The script needs to be fixed - or the .wast files need to be made available - right?  Where can I get them?
